packages/fluentui/react/src/components/Table/TableCell.tsx

```diff
@@ -1,108 +1,135 @@
 
 export interface TableCellProps extends UIComponentProps, ChildrenComponentProps, ContentComponentProps<ShorthandValue<BoxProps>> {
   /**
    * Accessibility behavior if overridden by the user.
    * @available TableCellBehavior
    * */
-  accessibility?: Accessibility;
+  accessibility?: Accessibility<TableCellBehaviorProps>;
 
   /**
    * Truncate cell's content
    */
   truncateContent?: boolean;
 }
 
+export type TableCellStylesProps = Pick<TableCellProps, 'truncateContent'>;
+
 export interface TableCellSlotClassNames {
   content: string;
 }
 
-class TableCell extends UIComponent<WithAsProp<TableCellProps>> {
-  static displayName = 'TableCell';
-
-  static className = 'ui-table__cell';
-
-  static slotClassNames: TableCellSlotClassNames = {
-    content: `${TableCell.className}__content`
-  };
-
-  static create: ShorthandFactory<TableCellProps>;
+const TableCell: React.FC<WithAsProp<TableCellProps>> &
+  FluentComponentStaticProps<TableCellProps> & {
+    slotClassNames: TableCellSlotClassNames;
+  } = props => {
+  const context: ProviderContextPrepared = React.useContext(ThemeContext);
+  const { setStart, setEnd } = useTelemetry(TableCell.displayName, context.telemetry);
+  setStart();
+  const cellRef = React.useRef<HTMLElement>();
+
+  const { children, content, truncateContent, className, design, styles, variables } = props;
+  const hasChildren = childrenExist(children);
+  const ElementType = getElementType(props);
+  const unhandledProps = getUnhandledProps(TableCell.handledProps, props);
+  const getA11yProps = useAccessibility(props.accessibility, {
+    debugName: TableCell.displayName,
+    actionHandlers: {
+      focusCell: e => {
+        e.preventDefault();
+        cellRef.current.focus();
+      },
+      performClick: e => {
+        handleClick(e);
+      }
+    },
+    rtl: context.rtl
+  });
 
-  static propTypes = {
-    ...commonPropTypes.createCommon({
-      content: false
+  const { classes, styles: resolvedStyles } = useStyles<TableCellStylesProps>(TableCell.displayName, {
+    className: TableCell.className,
+    mapPropsToStyles: () => ({
+      truncateContent
     }),
-    content: customPropTypes.every([
-      customPropTypes.disallow(['children']),
-      PropTypes.oneOfType([PropTypes.arrayOf(customPropTypes.nodeContent), customPropTypes.nodeContent])
-    ]),
-    truncateContent: PropTypes.bool
-  };
-
-  static defaultProps = {
-    accessibility: tableCellBehavior as Accessibility
-  };
-
-  cellRef = React.createRef<HTMLElement>();
-
-  actionHandlers = {
-    focusCell: e => {
-      e.preventDefault();
-      this.cellRef.current.focus();
-    },
-    performClick: e => {
-      this.handleClick(e);
-    }
-  };
+    mapPropsToInlineStyles: () => ({
+      className,
+      design,
+      styles,
+      variables
+    }),
+    rtl: context.rtl
+  });
 
-  handleClick = (e: React.SyntheticEvent) => {
+  const handleClick = (e: React.SyntheticEvent) => {
```

Comment:
```suggestion
  const handleClick = (e: React.MouseEvent | React.KeyboardEvent) => {
```

Let's be more explicit